File: golang/service.go
package golang

import "context"

type Service interface {
	Verify(ctx context.Context, code string) (string, error)
	Me(ctx context.Context, token string) (*User, error)
	ListProjects(ctx context.Context, token string) ([]Project, error)
	CreateProject(ctx context.Context, project *Project, token string) error
	UpdateProject(ctx context.Context, id string, project *Project, token string) error
	CreateResource(ctx context.Context, resource *Resource, token string) error
	DeleteResource(ctx context.Context, resourceID string, token string) error
}

File: rust/src/service.rs
use crate::error::Result;
use crate::types::{Resource, User};

#[async_trait::async_trait]
pub trait Service: Send + Sync {
    /// Verify authentication code and return access token
    async fn verify(&self, code: &str) -> Result<String>;

    /// Get current user information
    async fn me(&self, token: &str) -> Result<User>;

    /// Create a new resource
    async fn create_resource(&self, resource: &Resource, token: &str) -> Result<()>;

    /// Delete a resource by ID
    async fn delete_resource(&self, resource_id: &str, token: &str) -> Result<()>;
}
